rcl/src/rcl/logging_rosout.c

```diff
@@ -173,11 +173,15 @@ rcl_ret_t rcl_logging_rosout_init_publisher_for_node(
   const rosidl_message_type_support_t * type_support =
     rosidl_typesupport_c__get_message_type_support_handle__rcl_interfaces__msg__Log();
   rcl_publisher_options_t options = rcl_publisher_get_default_options();
-  // Late joining subscriptions get the last 10 seconds of logs, up to 1000 logs.
-  options.qos.depth = 1000;
-  options.qos.durability = RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL;
-  options.qos.lifespan.sec = 10;
-  options.qos.lifespan.nsec = 0;
+
+  // Late joining subscriptions get the user's setting of rosout qos options.
+  const rcl_node_options_t *node_options = rcl_node_get_options(node);
+  if (NULL == node_options){
+    RCL_SET_ERROR_MSG("node options was null.");
```

Comment:
I think that we do not need this, since error is already set.
https://github.com/ros2/rcl/blob/bfc0da993048ac81f2e1536d5e00a14722a4f0b9/rcl/src/rcl/node.c#L468-L469

Reply:
At first, i didn't output error message, just return error code.
But then i noticed the above [code](https://github.com/ros2/rcl/blob/master/rcl/src/rcl/logging_rosout.c#L155), so i added it.

Reply:
understood, but it will overwrite the error message. I do not see that is necessary.